### PR TITLE
BUG Fix versioned writes where subtables have no fields key

### PR DESCRIPTION
**Pull request by tractorcow**
Fixes some regressions in recent framework fixes to versioned.

Also remove unnecessary config nesting in tests which are now handled via core.

Fixes issue I mentioned at https://github.com/silverstripe-labs/silverstripe-fulltextsearch/pull/109#issuecomment-189010025

cc @dhensby , sorry it's late!

### CODE REVIEW

File: code/search/SearchUpdater.php [hunk 1]
```diff
@@ -90,14 +90,14 @@ public static function handle_manipulation($manipulation)
         $writes = array();
 
         foreach ($manipulation as $table => $details) {
-            if (!isset($details['id']) || !isset($details['fields'])) {
+            if (!isset($details['id'])) {
```
Comment (tractorcow) [inline code/search/SearchUpdater.php]: if `fields` is empty, we still need to run this loop to trigger the `update to more specific classname` line below.

**Comment by dhensby** [comment]
green ticks, yay!

**Comment by dhensby** [comment]
Let's tag a release?

**Comment by tractorcow** [comment]
Soon; I need to merge a variants fix first.

**Pull request merged**